cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -85,32 +95,15 @@ public DatasetSpecification create(Id.DatasetInstance datasetInstanceId, Dataset
       }
 
       DatasetSpecification spec = type.configure(datasetInstanceId.getId(), props);
+      if (props.getDescription() != null) {
+        spec = spec.setDescription(props.getDescription());
```

Comment:
I did this because I thought instance description should override description provided by type.